packages/@ngtools/json-schema/src/schema-class-factory.ts

```diff
@@ -132,7 +131,11 @@ class SchemaClassBase<T> implements SchemaClass<T> {
   /** Set a value from a JSON path. */
   $$set(path: string, value: any): void {
     const node = _getSchemaNodeForPath(this.$$schema(), path);
+
     if (node) {
+      if (node.constructor.name === 'EnumSchemaTreeNode' && node.items.indexOf(value) === -1) {
+        throw new InvalidUpdateValue('Invalid value can only be one of these: ' + node.items);
```

Comment:
@hansl Prints like this `Invalid value can only be one of these: Emulated,Native,None`